assigner/backends/decorators.py

```diff
@@ -0,0 +1,25 @@
+import logging
+
+from assigner.backends import GitlabBackend, MockBackend
+from assigner.config import requires_config
+
+
+logger = logging.getLogger(__name__)
+
+
+def requires_config_and_backend(func):
+    """Provides a backend depending on configuration."""
+    @requires_config
+    def wrapper(config, cmdargs, *args, **kwargs):
+        try:
+            config.backend
+        except KeyError:
```

Comment:
I think I'd rather propagate this `KeyError` out; if they haven't set a backend, none of this is going to work...